### PR TITLE
[RFC] Doc: autocmd for always switching to insert mode (:term)

### PR DESCRIPTION
**Pull request by mhinz**
As discussed on Gitter some minutes ago..

### CODE REVIEW

**Comment by fmoralesc** [comment]
Can this be in the main document for the terminal? I doubt anyone will see it in `various.txt`, especially if it's not tagged.

**Comment by mhinz** [comment]
You mean an own section in `nvim_terminal_emulator.txt`? Maybe `|nvim_terminal_emulator_usage|`?

File: runtime/doc/various.txt [hunk 1]
```diff
@@ -222,6 +222,11 @@ g8			Print the hex values of the bytes used in the
 			modified, but can be forced with "!".  See |termopen()|
 			and |nvim-terminal-emulator| for more information.
 
+			If you always want to switch to insert mode when
+			entering a terminal buffer, use this:
+>
+			      autocmd BufEnter term://* startinsert
```
Comment (justinmk) [inline runtime/doc/various.txt]: need more than BufEnter. E.g. it won't work for `<c-^>`.

Comment (mhinz) [inline runtime/doc/various.txt]: Uhm, that should work?

File: runtime/doc/various.txt [hunk 2]
```diff
@@ -222,6 +222,11 @@ g8			Print the hex values of the bytes used in the
 			modified, but can be forced with "!".  See |termopen()|
 			and |nvim-terminal-emulator| for more information.
 
+			If you always want to switch to insert mode when
```
Comment (ZyX-I) [inline runtime/doc/various.txt]: It is officially called terminal mode, not insert.

Comment (ZyX-I) [inline runtime/doc/various.txt]: You cannot switch to the insert mode in a terminal buffer at all.

Comment (mhinz) [inline runtime/doc/various.txt]: :+1:

**Comment by wsdjeg** [comment]
please do not use this autocmd,I will show you some bugs with airline.

**Comment by wsdjeg** [comment]
please see here    
https://github.com/vim-airline/vim-airline/issues/1029

File: runtime/doc/various.txt [hunk 3]
```diff
@@ -222,6 +222,11 @@ g8			Print the hex values of the bytes used in the
 			modified, but can be forced with "!".  See |termopen()|
 			and |nvim-terminal-emulator| for more information.
 
+			If you always want to switch to terminal mode when
+			entering a terminal buffer, use this:
```
Comment (justinmk) [inline runtime/doc/various.txt]: To switch to terminal-mode automatically:

Comment (mhinz) [inline runtime/doc/various.txt]: :+1:

**Pull request merged**